src/AutoBuild.jl

```diff
@@ -226,9 +226,8 @@ function build_tarballs(ARGS, src_name, src_version, sources, script,
             @info("Committing and pushing $(src_name)_jll.jl wrapper code version $(build_version)...")
         end
 
-        # For deploy discard build-only dependencies and transform in PackageSpec
+        # For deploy discard build-only dependencies
         filter!(dep -> !isa(dep, BuildDependency), dependencies)
```

Comment:
I think here we also need to force the eltype of the vector to be `Dependency`, `filter!` doesn't change it:
```suggestion
        # and make sure we get a `Vector{Dependency}`
        dependencies = Dependency[dep for dep in dependencies if !isa(dep, BuildDependency)]
```

Inspired by https://github.com/JuliaPackaging/Yggdrasil/commit/7474822ef4385f2c75982175245d4fee6312bcb3